tests/Testing/Fluent/AssertTest.php

```diff
@@ -271,6 +272,38 @@ public function testAssertWhereMatchesValue()
         $assert->where('bar', 'value');
     }
 
+    public function testAssertWhereFailsWithFloatInStrictMode()
+    {
+        $doubleValue = (float) 1234.0;
+
+        $stringifiedData = json_encode([
+            'bar' => $doubleValue,
+        ]);
+
+        $assertableJsonString = new AssertableJsonString($stringifiedData);
```

Comment:
If using AssertableJsonString in AssertTest is wrong, please let me know and I will refactor it.